### ISSUE TITLE
Service links aren't being created in Bridge mode

### ISSUE THREAD

**Issue by ajacques** [rancher/rancher#4387]
I recently upgrade from rancher server v1.9.x to v1.0.0. The upgrade wiped out my Rancher DB server so I had to start fresh. (Clean DB, no previous state).

Service links don't get created after upgrading to v1.0.0 which prevents any inter-container communication from happening when I use Bridge networking.

Rancher Version: v1.0.0

Docker Version: v1.6.2

OS: Ubuntu Server 3.13.0-83-generic # 127-Ubuntu SMP

Steps to Reproduce: 

* Create a new service stack: (Sample from a production .yml)
```
redis:
  cap_drop:
  - ALL
  labels:
    io.rancher.container.pull_image: always
  image: redis:3.0.7-alpine
  net: bridge
  mem_limit: 268435456
  cap_add:
  - CHOWN
  - SETGID
  - SETUID
WebUI:
  cap_drop:
  - ALL
  tty: true
  image: ubuntu:latest
  links:
  - redis:redis
  stdin_open: true
  net: bridge
```
* Login to the WebUI container with "Execute Shell"
* Execute `cat /etc/hosts` or `ping redis`

Results: 
The container 'redis' isn't resolvable

sudo docker inspect r-TestStack_WebUI_1:
```
[...]
    "HostConfig": {
[...]
        "IpcMode": "",
        "Links": null,
        "LogConfig": {
            "Config": null,
            "Type": "json-file"
        },
[...]
```
Redis NetworkSettings:
```
    "NetworkSettings": {
        "Bridge": "docker0",
        "Gateway": "172.17.42.1",
        "GlobalIPv6Address": "[---]:beef:231:359e:ae61",
        "GlobalIPv6PrefixLen": 80,
        "IPAddress": "172.17.0.134",
        "IPPrefixLen": 16,
        "IPv6Gateway": "fe80::1",
        "LinkLocalIPv6Address": "fe80::31:35ff:fe9e:ae61",
        "LinkLocalIPv6PrefixLen": 64,
        "MacAddress": "02:31:35:9e:ae:61",
        "PortMapping": null,
        "Ports": {
            "6379/tcp": null
        }
    },
```
WebUI NetworkSettings:
```
    "NetworkSettings": {
        "Bridge": "docker0",
        "Gateway": "172.17.42.1",
        "GlobalIPv6Address": "[---]:beef:231:351c:63a8",
        "GlobalIPv6PrefixLen": 80,
        "IPAddress": "172.17.0.135",
        "IPPrefixLen": 16,
        "IPv6Gateway": "fe80::1",
        "LinkLocalIPv6Address": "fe80::31:35ff:fe1c:63a8",
        "LinkLocalIPv6PrefixLen": 64,
        "MacAddress": "02:31:35:1c:63:a8",
        "PortMapping": null,
        "Ports": {}
    },
```

Expected:

I expect that the redis container is available as a link to the WebUI container. This is how it worked previously before I upgraded to Rancher v1.0.0. So far, no links appear to be created for any of my services in any of my stacks.

I've tried deleting and recreating the stack several times to see if that fixes anything, but so far nothing.

**Comment by deniseschannon** [comment]
Can you clarify some things on your versioning? 

What was your previous Rancher version? You mention v1.9.x, but everything before was v0.63.1 or v0.59.1, nothing close to v1.9.x. 

And are you saying your Docker version is actually 1.6.2? If so, we recommend upgrading to Docker 1.10.3 as that's what we do all of our testing. 

As for your specific issue, can you check that cross host communication is working. Can you exec into the network agent on 1 host and ping the IP (`10.42.x.x`) of another network agent? Also, you say that Rancher's upgrade wiped out your state so you had to start fresh, but you mention an upgraded setup. Is it a fresh install using the built in mysql server or where is the DB located?

**Comment by ajacques** [comment]
You're probably right. I'm not totally sure what the previous version was since I don't have the server container around on my host anymore. It would have been either v0.59.x or v0.63.x.

Basically doing an in-place upgrade keeping the Rancher server DB data around failed (This used to be an option that was mentioned in the upgrade docs) so I just deleted everything and started from scratch. That means brand new Rancher (no previous DB state) and wiped away all of the Rancher agent state data.

I'll see about upgrading Docker to 1.10.3 in a few minutes to try that out.

As of now, there is only one Rancher agent on this server so I can't ping any other agents and actually test anything useful since this is just a test environment.

Also I'm currently testing with the Bridge network configuration, not the Managed network configuration. Which as I understood it, did not use the Rancher network agent. I'm trying to start simple first.

From what I can tell, Rancher doesn't request that Docker create a container Link, which I expected since this was the Docker-based network connectivity.

tl;dr Fresh Rancher everything, single agent Rancher cluster, Non-Managed network configuration. Will try upgrading Docker to try that.

**Comment by ajacques** [comment]
After an eventful upgrade to 1.10, I had to again recreate my Rancher configuration + all containers. It appears to be generating the service Links correctly for Bridge networks.

I'm no-longer impacted by this issue, as such am going to resolve this issue. Thanks for your help and much appreciation on Rancher.
